tests/vm_backup_test.go

```diff
@@ -199,6 +199,70 @@ var _ = Describe("[smoke] VM Backup", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("started VM should be restored with new MAC address", func() {
+		// creating a started VM, so it works correctly also on WFFC storage
+		var err error
+		By("Starting a VM")
+		vm, err = framework.CreateStartedVirtualMachine(f.KvClient, f.Namespace.Name, framework.CreateVmWithGuestAgent("test-vm", f.StorageClass))
+		Expect(err).ToNot(HaveOccurred())
+
+		err = framework.WaitForVirtualMachineStatus(f.KvClient, f.Namespace.Name, vm.Name, kvv1.VirtualMachineStatusRunning)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Retrieving the original MAC address")
+		vm, err = f.KvClient.VirtualMachine(f.Namespace.Name).Get(context.TODO(), vm.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		originalMAC := vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+		if originalMAC == "" {
+			// This means there is no KubeMacPool running. We can simply choose a random address
+			originalMAC = "DE-AD-00-00-BE-AF"
+			update := func(vm *kvv1.VirtualMachine) *kvv1.VirtualMachine {
+				vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress = originalMAC
+				return vm
+			}
+			retryOnceOnErr(updateVm(f.KvClient, f.Namespace.Name, vm.Name, update)).Should(BeNil())
```

Comment:
maybe you should use patch here?

Reply:
This is a preexisting function used in several other tests, prefer to leave it here and consider refactoring the function itself in a follow-up.